### ISSUE TITLE
Inconsistency between @storybook/core-server buildDev & buildStatic options resolution (regression)

### ISSUE THREAD

**Issue by lihbr** [storybookjs/storybook#14561]
**Describe the bug**
`buildDev()` and `buildStatic()` don't resolve the `configDir` option the same way.

Basically `buildStatic()` now ignore provided `loadOptions.configDir` option.

This leads to build failing when using `buildStatic()` with a custom `configDir` provided through the `loadOptions` object.

* `buildDev()` configuration resolver: https://github.com/storybookjs/storybook/blob/next/lib/core-server/src/build-dev.ts#L122
* `buildStatic()` configuration resolver: https://github.com/storybookjs/storybook/blob/next/lib/core-server/src/build-static.ts#L110

Possible commit responsible for the regression (didn't read anything about that option being deprecated): https://github.com/storybookjs/storybook/commit/9bb89b7526c2d4d9687afe543bd59ac635a70347#diff-1f7816632a730834f23b99d2d097729e56e2401e30691f72e8a52b1a38b0cedfL259-R237

**To Reproduce**
n/a

See _Additional context_ below, sorry!

**Expected behavior**
`buildDev()` and `buildStatic()` should resolve the `configDir` directory consistently.

**Screenshots**
n/a

**Code snippets**
n/a

**System**

```
Environment Info:

  System:
    OS: Windows 10 10.0.19041
    CPU: (8) x64 Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
  Binaries:
    Node: 14.15.3 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - C:\Program Files\nodejs\yarn.CMD
    npm: 6.14.9 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 89.0.4389.114
    Edge: Spartan (44.19041.423.0), Chromium (89.0.774.75)
```

**Additional context**
Basically encountered this issue when using [@nuxtjs/storybook](https://github.com/nuxt-community/storybook) which uses a custom `configDir` (`.nuxt-storybook/storybook`) that is provided to both `buildDev()` and `buildStatic()` (see: [@nuxtjs/storybook/src/index.ts#L11-L18](https://github.com/nuxt-community/storybook/blob/master/src/index.ts#L11-L18)). Everything works fine with the dev server but fails when building static due to the above error.

Cheers! Let me know if anything~ Happy to provide a PR if confirmed valid but had trouble understanding the contributing guide~

cc farnabaz

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1199622255231368/1200183645922341) by [Unito](https://www.unito.io/learn-more)

**Comment by farnabaz** [comment]
This is also reported in nuxt-community/storybook#259

**Comment by ndelangen** [comment]
@lihbr I think your analysis is spot on!

https://github.com/storybookjs/storybook/blob/next/lib/core-server/src/build-static.ts#L110

Should be changed! If you are to open a PR, I'm sure we can get it merge today! 🙇

**Comment by lihbr** [comment]
Awesome! Can you just please highlight me as to which source/target branch should I use? Contributing guide got me confused :/ Will try to hop on Discord~

**Comment by ndelangen** [comment]
You should branch off (and target) The `next` branch

**Comment by shilman** [comment]
Zoinks!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.3.0-alpha.6 containing PR #14566 that references this issue. Upgrade today to the `@next` NPM tag to try it out!

```
npx sb upgrade --prerelease
```

Closing this issue. Please re-open if you think there's still more to do.

**Comment by shilman** [comment]
Ermahgerd!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.2.8 containing PR #14566 that references this issue. Upgrade today to the `@latest` NPM tag to try it out!

```
npx sb upgrade
```
